modules/programs/ghostty.nix

```
@@ -0,0 +1,171 @@
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+
+    enableFishIntegration = lib.mkEnableOption ''
+      fish shell integration.
+
+      This is ensures that shell integration works in more scenarios, such as switching shells within Ghostty.
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+
+    enableZshIntegration = lib.mkEnableOption ''
+      zsh shell integration.
+
+      This is ensures that shell integration works in more scenarios, such as switching shells within Ghostty.
+      But it is not needed to have shell integration.
+      See <https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup> for more information
+    '';
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      programs.ghostty.settings = lib.mkIf cfg.clearDefaultKeybinds {
+        keybind = lib.mkBefore [ "clear" ];
+      };
+
+      xdg.configFile = lib.mkMerge [
+        {
+          "ghostty/config" = lib.mkIf (cfg.settings != { }) {
+            source = keyValue.generate "ghostty-config" cfg.settings;
+          };
+        }
+
+        (lib.mkIf (cfg.themes != { }) (lib.mapAttrs' (name: value: {
+          name = "ghostty/themes/${name}";
+          value.source = keyValue.generate "ghostty-${name}-theme" value;
+        }) cfg.themes))
+      ];
+    }
+
+    (lib.mkIf cfg.installVimSyntax {
+      programs.vim.plugins = [ cfg.package.vim ];
+    })
+
+    (lib.mkIf cfg.installBatSyntax {
+      programs.bat = {
+        syntaxes.ghostty = {
+          src = cfg.package;
+          file = "share/bat/syntaxes/ghostty.sublime-syntax";
+        };
+        config.map-syntax = [ "*/ghostty/config:Ghostty Config" ];
+      };
+    })
+
+    (lib.mkIf cfg.enableBashIntegration {
+      # Make order 101 to be placed exactly after bash completions, as Ghostty documentation suggests
+      # sourcing the script as soon as possible
+      programs.bash.initExtra = lib.mkOrder 101 ''
+        if [[ -n "''${GHOSTTY_RESOURCES_DIR}" ]]; then
+          builtin source "''${GHOSTTY_RESOURCES_DIR}/shell-integration/bash/ghostty.bash"
+        fi
+      '';
+    })
+
+    (lib.mkIf cfg.enableFishIntegration {
+      programs.fish.shellInit = ''
+        if set -q GHOSTTY_RESOURCES_DIR
+          source "$GHOSTTY_RESOURCES_DIR/shell-integration/fish/vendor_conf.d/ghostty-shell-integration.fish"
+        end
+      '';
+    })
+
+    (lib.mkIf cfg.enableZshIntegration {
+      programs.zsh.initExtra = ''
+        if [[ -n $GHOSTTY_RESOURCES_DIR ]]; then
+          source $GHOSTTY_RESOURCES_DIR/shell-integration/zsh/ghostty-integration
```

```suggestion
          source "$GHOSTTY_RESOURCES_DIR/shell-integration/zsh/ghostty-integration"
```
Turns out zsh can have issues with the unquoted value in certain cases.

---

Updated.